src/scripts/utils/debounce.ts

```diff
@@ -0,0 +1,12 @@
+export function debounce<T extends (...args: Parameters<T>) => void>(
+    func: T,
+    delay: number
+) {
+    let timer: ReturnType<typeof setTimeout> | null = null;
+    return (...args: Parameters<T>) => {
+        if (timer) clearTimeout(timer);
+        timer = setTimeout(() => {
+            func(...args);
```

Comment:
Can attach a `this` along with args.
```suggestion
            func.apply(this, ...args);
```

Reply:
this is showing error that this has type `any`

Reply:
You need to provide a type for it.

Reply:
Fixed (https://github.com/krushna-simform/product-management-system/pull/3/commits/0c5a0eeb6fd1c2cb53cce045dc1155266754bbab)

Reply:
@krushna-simform can we use `ThisParameterType<T>`?
- https://www.typescriptlang.org/docs/handbook/utility-types.html#thisparametertypetype

Reply:
Yes, we can i will try to implement that.